curves/bls12_381/src/curves/g1.rs

```diff
@@ -144,36 +145,33 @@ impl<H: HostFunctions> SWCurveConfig for Config<H> {
         bases: &[Affine<Self>],
         scalars: &[<Self as CurveConfig>::ScalarField],
     ) -> Result<Projective<Self>, usize> {
-        let bases: Vec<Vec<u8>> = bases.iter().map(|elem| serialize_argument(*elem)).collect();
-        let scalars: Vec<Vec<u8>> = scalars
-            .iter()
-            .map(|elem| serialize_argument(*elem))
-            .collect();
+        let bases: ArkScale<&[Affine<Self>]> = bases.into();
+        let scalars: ArkScale<&[<Self as CurveConfig>::ScalarField]> = scalars.into();
 
-        let result = H::bls12_381_msm_g1(bases, scalars);
+        let result = H::bls12_381_msm_g1(bases.encode(), scalars.encode()).unwrap();
 
-        let result = deserialize_result::<Affine<Self>>(&result);
-        Ok(result.into())
+        let result = <ArkScale<Projective<Self>> as Decode>::decode(&mut result.clone().as_slice());
```

Comment:
`ark_scale::hazmat::ArkScaleProjective<Projective<Self>>::`

Reply:
This was done. Probably forgot to push.